libafl/src/executors/timeout.rs

```diff
@@ -151,22 +170,38 @@ where
         unsafe {
             let data = &mut GLOBAL_STATE;
             write_volatile(
-                &mut data.timer_queue,
-                &mut self.timer_queue as *mut _ as *mut c_void,
+                &mut data.tp_timer,
+                &mut self.tp_timer as *mut _ as *mut c_void,
             );
-            let code = CreateTimerQueueTimer(
-                &mut self.ph_new_timer,
-                &self.timer_queue,
-                Some(self.executor.timeout_handler()),
-                &mut GLOBAL_STATE as *mut _ as *mut c_void,
-                self.milli_sec,
-                0,
-                WORKER_THREAD_FLAGS::default(),
+            write_volatile(
+                &mut data.critical,
+                &mut self.critical as *mut _ as *mut c_void,
             );
-            if !code.as_bool() {
-                return Err(Error::Unknown("CreateTimerQueue failed.".to_string()));
-            }
+            write_volatile(
+                &mut data.timeout_input_ptr,
```

Comment:
Here I copy `current_input_ptr` to `timeout_input_ptr`
I have to do this because `run_target` clears `current_input_ptr` to `null` before it enters the critical section.

I wonder if we can put the pre_exec/post_exec back at least for the `Executor`?
then we don't need this copy